core/src/main/java/io/aiven/klaw/controller/TopicController.java

```diff
@@ -91,19 +91,30 @@ public ResponseEntity<Map<String, String>> getTopicTeam(
         topicControllerService.getTopicTeamOnly(topicName, patternType), HttpStatus.OK);
   }
 
-  /*
-     For executing topic requests
-  */
+  /**
+   * @param pageNo Which page would you like returned e.g. 1
+   * @param currentPage Which Page are you currently on e.g. 1
+   * @param requestsType What type of requests are you looking for e.g. 'created' or 'deleted'
+   * @param teamName The name of the team that that you wish to filter the results by, e.g.
+   *     'Octopus'
+   * @param env The name of the environment you would like returned e.g. '1' or '4'
+   * @param search A wildcard search term that searches topicNames.
+   * @return A List of Topic Requests filtered by the provided parameters.
+   */
   @RequestMapping(
       value = "/getTopicRequestsForApprover",
       method = RequestMethod.GET,
       produces = {MediaType.APPLICATION_JSON_VALUE})
   public ResponseEntity<List<TopicRequestModel>> getTopicRequestsForApprover(
       @RequestParam("pageNo") String pageNo,
       @RequestParam(value = "currentPage", defaultValue = "") String currentPage,
-      @RequestParam(value = "requestsType", defaultValue = "created") String requestsType) {
+      @RequestParam(value = "requestsType", defaultValue = "created") String requestsType,
+      @RequestParam(value = "teamName", required = false) String teamName,
```

Comment:
request raised by this 'teamName' here or assigned to?

Reply:
for topic requests, its the same team who approves all the requests, except claim requests. So I assume this teamName filter is specifically for claim requests, and rest all the requests are by the same team as the approver

Reply:
Raised by I will add this to the javadocs.